File: C programming/Competitive/Assignments/Assignment_12/Que_1.c
//question 1: Write a program which accepts a number from user and display all digits in reverse order.

#include<stdio.h>

void DisplayDigit(int iNo)
{
    int iDigit = 0;

    if(iNo < 0)   // handle negative input
    {
        iNo = -iNo;
    }

    while(iNo != 0)
    {
        iDigit = iNo % 10;     // extract last digit
        printf("%d\n", iDigit);
        iNo = iNo / 10;        // remove last digit
    }
}

int main()
{
    int iValue = 0;

    printf("Enter number: ");
    scanf("%d", &iValue);

    DisplayDigit(iValue);

    return 0;
}

File: Java Programming/Competitve/Assignments/Assignment_21/Que_1.java
//write a program to calculate the product of a digit of a number 

class Logic
{
    void productOfDigits(int num)
    {
        if (num == 0)
         {
            System.out.println(0);
            return;
        }

        // Handle negative numbers
        if (num < 0) 
        {
            num = -num;
        }

        int product = 1;

        while (num > 0)
         {
            int digit = num % 10;
            product *= digit;
            num /= 10;
        }

        System.out.println(product);
    }
}

class Que_1
{
    public static void main(String[] args)
    {
        Logic obj = new Logic();
        obj.productOfDigits(234);
        
    }
}


